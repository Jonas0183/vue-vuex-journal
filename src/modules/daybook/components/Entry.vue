<template>
    <div
        class="mb-3 entry-container  pointer p-2"
        @click="$router.push( { name: 'entry', params: { id:10 } } )">
        <!-- titulo -->
        <div
            class="entry-title d-flex">
            <span class="text-success fs-5 fw-bold">15 </span>
            <span class="mx-1 fs-5">Julio </span>
            <span class="mx-2 fw-light">2021, Jueves</span>
        </div>
        <div class="entry-description">
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Earum rem iusto dolore minus accusamus et reprehenderit laboriosam itaque numquam labore? Quisquam ipsum reprehenderit esse, voluptate dolorum modi ab unde consectetur?
        </div>
    </div>
</template>
<script>
export default {
    
}
</script>
<style lang="scss" scoped>
.entry-container{
    border-bottom:1px solid #2c3e50;
    transition: 0.2 ease-in;
    &:hover{
        background-color: lighten($color: grey, $amount: 45);
        transition: 0.2 ease-in;

    }
.entry-description{
    font-size: 12px;
}
}
</style>