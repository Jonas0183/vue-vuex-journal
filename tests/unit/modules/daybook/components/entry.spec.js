import { shallowMount } from "@vue/test-utils";
import { journalState } from "../../..//mock-data/test-journal-state";

import Entry from "@/modules/daybook/components/Entry";

describe("Pruebas en el entry component", () => {
  const mockRouter = {
    push: jest.fn(),
  }

  const wrapper = shallowMount(Entry, {
    props: {
      entry: journalState.entries[0],
    },
    global: {
        mocks:{
            $router: mockRouter,
        }

    },
  });

  test("Debe hacer match con el snapshot", () => {
    
    expect(wrapper.html()).toMatchSnapshot();
  
  });

  test("Debe redireccionar al hacer click en el entry-container", () => {});

  const entryContainer = wrapper.find('.entry-container')
  entryContainer.trigger('click')

  expect( mockRouter.push ).toHaveBeenCalledWith({
      name: 'entry',
      params: {
          id: journalState.entries[0].id
      }
  })

  test("Pruebas en las propiedades computadas", () => {
    
    expect( wrapper.vm.day ).toBe(26)
    expect( wrapper.vm.month ).toBe('Mayo')
    expect( wrapper.vm.yearDay ).toBe(' 2022, Jueves ')
  });
});
