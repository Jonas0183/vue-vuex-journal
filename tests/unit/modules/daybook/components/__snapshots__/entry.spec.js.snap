// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el entry component Debe hacer match con el snapshot 1`] = `
<div class="mb-3 entry-container pointer p-2">
  <!-- titulo -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">26</span><span class="mx-1 fs-5">Mayo</span><span class="mx-2 fw-light"> 2022, Jueves </span></div>
  <div class="entry-description">hola mundo mock</div>
</div>
`;
